src/test/java/spoon/test/filters/FilterTest.java

```diff
@@ -115,9 +115,17 @@ public void testNameFilter() throws Exception {
 		CtClass<?> foo = factory.Package().get("spoon.test.filters.testclasses").getType("Foo");
 		assertEquals("Foo", foo.getSimpleName());
 		List<CtNamedElement> elements = foo.getElements(new NameFilter<>("i"));
+		// contract: tests NameFilter constructor for cases other than null
 		assertEquals(1, elements.size());
+		// contract: tests NameFilter constructor for null value case
```

Comment:
The contract should not say what the test does, but what is expected to happen. For example, this one ought to say `contract: NameFilter constructor should throw IllegalArgumentException when passed null`

The same problem exists with all of the contracts you have written here, they just say what the test does, while they should detail the expected outcome of some action.

Reply:
Got it 👍🏼